### PR TITLE
`overrides` and `namespaced` functions

### PR DESCRIPTION
**Pull request by koresar**
Please, take a good look at the README. :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,3 +102,51 @@ const InstanceLogger = init(({ name }) => {
 
 const george = InstanceLogger({ name: 'George' }); // Created new instance: George
 ```
+
+
+### overrides()
```
Comment (FredyC) [inline README.md]: Perhaps add here a link to [article](https://medium.com/@koresar/fun-with-stamps-episode-3-comparing-with-the-es2015-classes-e387ef041896) to show a real use case.

File: src/index.js [hunk 2]
```diff
@@ -1,5 +1,10 @@
 import isFunction from 'lodash/isFunction';
 import isObject from 'lodash/isObject';
+import assign from 'lodash/assign';
```
Comment (danielkcz) [inline src/index.js]: Once again, why not just import from `lodash`?

File: src/test/init.js [hunk 3]
```diff
@@ -26,7 +62,7 @@ test('init()', nest => {
     const mutators = ['a', 'b', 'c'].map(
       (i) => (o, { instance }) => { instance[i] = i; });
 
-    const actual = init(...mutators)();
+    const actual = init(mutators[0], mutators.slice(1, 3))();
```
Comment (ericelliott) [inline src/test/init.js:65]: Why is this change needed?

File: README.md [hunk 4]
```diff
@@ -102,3 +102,52 @@ const InstanceLogger = init(({ name }) => {
...
```
Comment (ericelliott) [inline README.md]: This seems like a confusing name for the described functionality. I'll need to read the blog post for more context so I can make a more useful comment.

Comment (koresar) [inline README.md]: Not sure which blog post you're referencing. But the `overrides` is genuinely the name you came up with here: https://github.com/stampit-org/stamp-utils/issues/1#issuecomment-145359335

I'm open to suggestions because it feels like `overrides` is to generic name.

File: src/test/init.js (hunk 3 above)
Comment (koresar) [inline src/test/init.js:65]: So that people can provide initializers not just as functions `init(f1, f2)` but also as an array of functions: `init([f1, f2])`, or even mix of those `init(f1, [f2, f2])`.
See README changes.

File: src/index.js (hunk 2 above)
Comment (koresar) [inline src/index.js]: Because most bundlers do "tree shaking" by file names currently. But I'm fine to change it to `import _ from 'lodash';` because there is the babel plugin (which you showcased in stampit) which does lodash tree shaking. :)
I'll do that same change to the `compose.js` file too.

**Comment by koresar** [comment]
@FredyC thanks for the idea to reference the article. Although, this module lacks `methods` utility to be fully article compliant so to say. :)
I'll add `methods` now.

**Comment by koresar** [comment]
This PR is ready. Please, take a look again.

File: README.md (hunk 4 above)
Comment (danielkcz) [inline README.md]: Frankly I had also bit trouble understanding what is it suppose to do till I found that article. Calling it `overrides` might be misleading because you can always compose another initializer that would override instance values with something else. Seeing `overrides` I would assume that it has a priority over anything else. Instead of a _fancy_ name I would go for something like `initialAssign`. I see that @ericelliott was against name `assignAllOptions` in a referenced comment, but I don't like that one in particular either.

File: src/test/init.js (hunk 3 above)
Comment (danielkcz) [inline src/test/init.js:65]: Why is the `init` even here? This seems to be clashing with `stampit.init`. I think there should be clean separation of what are `stamp-utils` for, not to have a different versions of a same thing. Thus I recommend removing `init` here and instead referencing `stampit` in docs. Let's leave `stamp-utils` as extra helper functions.

File: README.md (hunk 4 above)
Comment (ericelliott) [inline README.md]: I think I misunderstood the implementation you had in mind. Maybe `instanceProps`?

Comment (koresar) [inline README.md]: `assignOptions`? Since we are _assigning options_ object.

File: src/test/init.js (hunk 3 above)
Comment (koresar) [inline src/test/init.js:65]: We can have both - `init` in `stamp-utils`, and `init` in `stampit`. They would just do different things.

This module is, as declared, a set of helper functions. No more. Each function receives few arguments and always returns a simplest possible stamp (no added statics, or configuration, or else). So, `stamp-utils` is a minimalistic KISS kind of module. All functions are pure. Like `left-pad`.

Whereas, `stampit` is a full-featured "utility-belt" kind of module. It infects your stamps, adds a zillion statics (including `.init()`), and tries to be "everything you'd every need". Like `lodash`.

Would you agree with such evolution directions?

Comment (danielkcz) [inline src/test/init.js:65]: To me personally it's super confusing to have a same name of a function with different behavior. I already said it with stampit, but having X different implementations of `compose` seems like a way to unmaintainable hell in my opinion. Call me lazy, but I think this is a bad decision.

Just think of a new user coming to stamps which are already mind blowing on its own. That user has to face decision what module to pick: stamp-specification, stampit or stamp-utils? Studying what are differences and what he might actually need. I believe that for such innovative approach, the entry should be as smooth as possible.

I think that eventually there should be just single most performant implementation of a `compose` and ideally any helper functions could be made as a separate module (I really like babel plugins way).

File: README.md (hunk 4 above)
Comment (danielkcz) [inline README.md]: I would avoid word `options` as not everyone is aware that first argument is called like that. Using `props` as a part of name feels weird too since it's mostly term introduced by `stampit`. I think it should surely contain word _assign_ so it's clear what it does, perhaps `assignInstance` or even `assignToInstance` or `instanceAssign`.

Comment (ericelliott) [inline README.md]: `assignToInstance()` is less confusing, I think. =)

Comment (koresar) [inline README.md]: Best choice! I'll use that name. Expect more commits to this PR. And thanks guys!

File: src/test/init.js (hunk 3 above)
Comment (koresar) [inline src/test/init.js:65]: 100% agree. Thank you for binging the simplification topic to the surface. We somehow missed that point.
I created the new issue for discussion here: https://github.com/stampit-org/stampit/issues/221

**Comment by koresar** [comment]
@ericelliott feel free to publish.
